frontend/src/metabase/query_builder/components/filters/modals/InlineDatePicker/InlineDatePicker.unit.spec.tsx

```diff
@@ -0,0 +1,369 @@
+        { include_current: false },
+      ],
+      null,
+      query,
+    );
+    const changeSpy = jest.fn();
+    const clearSpy = jest.fn();
+
+    render(
+      <InlineDatePicker
+        query={query}
+        filter={testFilter}
+        newFilter={newFilter}
+        dimension={dateDimension}
+        onChange={changeSpy}
+        onClear={clearSpy}
+      />,
+    );
+
+    expect(screen.getByText("Yesterday")).toHaveAttribute(
+      "aria-selected",
+      "true",
+    );
+    // clicking a selected shortcut value again de-selects it and clears the filter
+    screen.getByText("Yesterday").click();
+
+    expect(changeSpy).not.toHaveBeenCalled();
+    expect(clearSpy).toHaveBeenCalledTimes(1);
+  });
+
+  it.skip("adds a custom value", async () => {
+    const changeSpy = jest.fn();
+
+    render(
+      <InlineDatePicker
+        query={query}
+        filter={undefined}
+        newFilter={newFilter}
+        dimension={dateDimension}
+        onChange={changeSpy}
+        onClear={changeSpy}
+      />,
+    );
+
+    screen.getByLabelText("more options").click();
+    await waitFor(() => screen.getByText("Relative dates..."));
+    screen.getByText("Relative dates...").click();
+    await waitFor(() => screen.getByTestId("relative-datetime-value"));
+
+    const input = screen.getByTestId("relative-datetime-value");
+    userEvent.clear(input);
+    userEvent.type(input, "88");
+    // FIXME: for some reason this button never gets enabled
+    await waitFor(() =>
+      expect(screen.getByText("Add filter")).not.toBeDisabled(),
+    );
+    screen.getByText("Add filter").click();
+    await waitFor(() => expect(changeSpy).toHaveBeenCalled());
+
+    expect(changeSpy).toHaveBeenCalledWith([
+      "time-interval",
+      ["field", dateField.id, null],
+      -88,
+      "day",
+    ]);
+  });
+
+  it.skip("changes a custom value", async () => {
+    const testFilter = new Filter(
+      ["between", ["field", dateField.id, null], "1605-11-05", "2005-11-05"],
+      null,
+      query,
+    );
+
+    const changeSpy = jest.fn();
+
+    render(
+      <InlineDatePicker
+        query={query}
+        filter={testFilter}
+        newFilter={newFilter}
+        dimension={dateDimension}
+        onChange={changeSpy}
+        onClear={changeSpy}
+      />,
+    );
+
+    const btn = screen.getByText("between November 5, 1605 November 5, 2005");
+    userEvent.click(btn);
+    await waitFor(() => screen.getByDisplayValue("11/05/1605"));
+    const input = screen.getByDisplayValue("11/05/1605");
+
+    userEvent.clear(input);
+    userEvent.type(input, "09/05/1995");
+
+    // FIXME: for some reason this button never gets enabled
+    await waitFor(() =>
+      expect(screen.getByText("Update filter")).not.toBeDisabled(),
+    );
+
+    screen.getByText("Update filter").click();
+
+    expect(changeSpy).toBeCalledWith([
+      "between",
+      ["field", dateField.id, null],
+      "1995-09-05",
+      "2005-11-05",
+    ]);
+  });
```

Comment:
@nemanjaglumac I was hoping you or someone else with a bit better knowledge of jest + testing library could help me get this test working properly.  Unfortunately, I cannot get the date picker component's internal state to update in order to commit the filter in this test.

It looks like the new date picker does not have dedicated unit tests, and I'm wondering if this is why?

Reply:
@iethree I vaguely remember some problems I had with the date picker in unit tests, but that was 1-1,5 years ago! A lot has changed since then (or at least I hope it has). Seems like this would be better answered by someone from the FE team who has experience with this component. cc @kulyk or @alxnddr (if you guys didn't work with date picker, please tag someone else if you they have).

Reply:
I think @akiselev worked on the new date picker?